### PR TITLE
Updated testing and 1.14 requirements

### PR DESCRIPTION
**Pull request by brtnfld**
updated from 1.13 HDF5 to 1.14 requirements
* 1.13 is not a release that should be used; moved using a minimum of 1.14 and changed code to reflect this.
* removed hardcoded version of HDF5 targets to use; always uses the latest release.

### CODE REVIEW

File: doc/log_cache_async_vol.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 * [Build Instructions](#build-instructions)
 * [Run Instructions](#run-e3sm-io)
 
-This demo uses [E3SM-IO](https://github.com/Parallel-NetCDF/E3SM-IO) to show how to run the Log VOL connector on top of the [Cache VOL connector](https://github.com/hpc-io/vol-cache) and the [Async VOL connector](https://github.com/hpc-io/vol-async).
+This demo uses [E3SM-IO](https://github.com/Parallel-NetCDF/E3SM-IO) to show how to run the Log VOL connector on top of the [Cache VOL connector](https://github.com/hpc-io/vol-cache) and the [Async VOL connector](https://github.com/HDFGroup/vol-async).
```
Comment (jhendersonHDF) [inline doc/log_cache_async_vol.md:5]: Does the link to the cache VOL need to be updated here too?

File: src/H5VL_log_dataset.cpp [hunk 2]
```diff
@@ -387,18 +377,32 @@ herr_t H5VL_log_dataset_read_2 (size_t count,
                                 void *buf[],
                                 void **req) {
     herr_t err = 0;
+    H5VL_log_dset_t *dp       = (H5VL_log_dset_t *)dset;
+    H5VL_log_dset_info_t *dip = NULL;  // Dataset info
+    hid_t dsid;                        // Dataset space id
+    H5VL_log_selections *dsel = NULL;  // Selection blocks
```
Comment (jhendersonHDF) [inline src/H5VL_log_dataset.cpp:383]: It looks like all these new variables above are unused here.

**Review by jhendersonHDF — APPROVED** [review]

File: doc/log_cache_async_vol.md (hunk 1 above)
Comment (brtnfld) [reply doc/log_cache_async_vol.md]: Fixed and also updated "DataLib-ECP" references

File: src/H5VL_log_dataset.cpp (hunk 2 above)
Comment (brtnfld) [reply src/H5VL_log_dataset.cpp]: Fixed.

**Review by wkliao — APPROVED** [review]
Looks good to me.

**Pull request merged**